### PR TITLE
SQL refactoring

### PR DESCRIPTION
**Pull request by ImeevMA**
Minor refactoring of the SQL code.

### CODE REVIEW

**Review by Gerold103 — APPROVED** [review]
Thanks for the patch!

File: src/box/sql/prepare.c [hunk 1]
```diff
@@ -163,26 +163,23 @@ sql_stmt_compile(const char *zSql, int nBytes, struct Vdbe *pReprepare,
 	return rc;
 }
 
-/*
- * Rerun the compilation of a statement after a schema change.
- */
 int
-sqlReprepare(Vdbe * p)
+sqlReprepare(struct Vdbe *v)
```
Comment (locker) [inline src/box/sql/prepare.c:167]: Why do you rename `p` to `v`? It bloats the diff for no reason. The commit message says nothing about this change.

File: src/box/sql/vdbe.c [hunk 2]
```diff
@@ -4460,7 +4457,6 @@ default: {          /* This is really OP_Noop and OP_Explain */
 
 	/* This is the only way out of this procedure. */
 vdbe_return:
-	p->aCounter[SQL_STMTSTATUS_VM_STEP] += (int)nVmStep;
```
Comment (locker) [inline src/box/sql/vdbe.c:4463]: Looks unrelated.

File: src/box/sql/prepare.c (hunk 1 above)
Comment (ImeevMA) [reply src/box/sql/prepare.c:167]: Should I add this to the description? In this patch, I changed the use of the `struct Vdbe`, so I decided to rename variables in some places from `p` (it seems to be short for `pointer`) to the more specialized `v` (short for `vdbe`).

File: src/box/sql/vdbe.c (hunk 2 above)
Comment (ImeevMA) [reply src/box/sql/vdbe.c:4463]: This field is used only in the `sql_stmt_status()` function, which I decided to remove instead of refactoring, since it is not used anywhere.

Comment (locker) [reply src/box/sql/vdbe.c:4463]: It'd be better to drop `sql_stmt_status` in a separate commit then, because this change is unrelated to dropping the `sql_stmt` alias.

File: src/box/sql/prepare.c (hunk 1 above)
Comment (locker) [reply src/box/sql/prepare.c:167]: I don't think you should change local variable names, because `p` sounds fine to me as a local variable of type `Vdbe`. Moreover, even after this patch is applied, it's still used (e.g. in `sqlVdbeExec`). That is, this change doesn't make the code look any better - it only bloats the diff and makes it more difficult to review.

Comment (ImeevMA) [reply src/box/sql/prepare.c]: Fixed. Moved all dropped functions in the new commit.

Also, I will try to stick to the "one change = one commit" rule in the future, even with refactoring.

File: src/box/sql/vdbe.c (hunk 2 above)
Comment (ImeevMA) [reply src/box/sql/vdbe.c:4463]: Fixed. Moved all dropped functions in the new commit.

Also, I will try to stick to the "one change = one commit" rule in the future, even with refactoring.

**Review by locker — APPROVED** [review]
